litestar/routes/http.py

```diff
@@ -86,8 +86,10 @@ async def handle(self, scope: HTTPScope, receive: Receive, send: Send) -> None:
         if after_response_handler := route_handler.resolve_after_response():
             await after_response_handler(request)
 
+        if request._form is not Empty:
+            await request._form.close()
         if form_data := scope.get("_form", {}):
-            await self._cleanup_temporary_files(form_data=cast("dict[str, Any]", form_data))
+            await FormMultiDict.from_form_data(cast("dict[str, Any]", form_data)).close()
```

Comment:
I'm not sure I understand why we need both. In case both of these `_form`s are set, they should point to the same thing / contain the same `UploadFile` instances. If that's not the case, there's a bug somewhere

Reply:
I don't know but if you comment out either of them some tests fail (different tests for each `_form`).

Reply:
I think it's a bug. I'm investigating

Reply:
I was curious about the pytest-asyncio bug and removed that dependency to only use anyio and found another unclosed ressource that way:

![20241129_1357_1620x210_1732885025](https://github.com/user-attachments/assets/2c664f83-11bc-4824-9bf3-0c8776b0d8a9)

i'm going to make it a draft PR so that maybe you can get some parts of it, dealing with ResourceWarning is very tedious as the reporting of the error is often completely different fro where it happens

Reply:
Got it. 

There were 2 thing:

- We created an `UploadFile` manually in a test and didn't close it
- When handling multipart extraction, an exception during parsing could result in files not being closed